src/DataValues/TimeValueCalculator.php

```diff
@@ -125,4 +185,115 @@ public function getSecondsForPrecision( $precision ) {
 		throw new InvalidArgumentException( "Unable to get seconds for precision $precision." );
 	}
 
+	/**
+	 * @param $timestamp
+	 * @param $precision
+	 *
+	 * @return string
+	 */
+	private function timestampAbsFloor( $timestamp, $precision ) {
+		// The year is padded with zeros to have 16 digits
+		$timestamp = substr_replace(
+			$timestamp,
+			str_repeat( '0', $this->MAX_LENGTH_TIMESTAMP - strlen( $timestamp ) ),
+			1,
+			0
+		);
+		$numCharsToModify = $this->charsAffectedByPrecision( $precision );
+		$timestamp = substr( $timestamp, 0, -$numCharsToModify ) .
+			substr( $this->TIMESTAMP_ZERO, -$numCharsToModify );
+		return $timestamp;
+	}
+
+	/**
+	 * @param $timestamp
+	 * @param $precision
+	 *
+	 * @return string
+	 */
+	private function timestampAbsCeiling( $timestamp, $precision ) {
+		// The year is padded with zeros to have 16 digits
+		$timestamp = substr_replace(
+			$timestamp,
+			str_repeat( '0', $this->MAX_LENGTH_TIMESTAMP - strlen( $timestamp ) ),
+			1,
+			0
+		);
+		$numCharsToModify = $this->charsAffectedByPrecision( $precision );
+		// WARNING: Day 31 will be applied to all months
+		$timestampCeiling = substr( $timestamp, 0, -$numCharsToModify ) .
+			substr( $this->HIGHEST_TIMESTAMP, -$numCharsToModify );
+		if ( $precision === TimeValue::PRECISION_MONTH ) {
```

Comment:
Please add test cases for this month-specific code path.

Reply:
There's a test for this in testGetLowerTimestamp() and another in testGetHigherTimestamp(). But yes, it would be good to have more, I'll add them.